### PR TITLE
Desktop logout link

### PR DESCRIPTION
**Pull request by lemald**
Asana ticket: [⚙️ Add log out button to desktop top nav](https://app.asana.com/0/1148853526253420/1206526459943495/f)

### CODE REVIEW

File: assets/tests/components/nav/topNav.test.tsx
Comment (firestack) [inline assets/tests/components/nav/topNav.test.tsx:99]: suggestion: could we use the testing library `toHaveAttribute` matcher here?

File: assets/tests/components/nav/topNav.test.tsx
Comment (firestack) [inline assets/tests/components/nav/topNav.test.tsx:96]: nit(non-blocking): I'd prefer this to be `toBeVisible()`, it's essentially like `toBeInTheDocument`, but if we ever get CSS into our jest tests, we'll get better tests for "free".
```suggestion
      expect(logoutLink).toBeVisible()
```

File: assets/css/nav/_top_nav.scss
Comment (lemald) [reply assets/css/nav/_top_nav.scss:67]: I missed that note from design - was that just for the internal divider or also the border around the dropdown?

File: assets/tests/components/nav/topNav.test.tsx
Comment (firestack) [reply assets/tests/components/nav/topNav.test.tsx:99]:
```suggestion
      expect(logoutLink).toHaveAttribute("href", "/auth/keycloak/logout")
```

Comment (lemald) [reply assets/tests/components/nav/topNav.test.tsx]: Heh I figured there was a better way to do this that I was forgetting. 1326f7dd

File: assets/css/nav/_top_nav.scss
Comment (firestack) [reply assets/css/nav/_top_nav.scss:67]: I don't think it was written down, but as I understood it, we're likely going to set bootstraps `$border-color` to `tokens_2024.$gray-400`, which would apply it to the `<hr/>` as well. I don't think it's worth guessing what we'll do in the future though, it's probably better to stick to defaults and wait for the reboot/typo/colors work to be done.

File: assets/tests/components/nav/topNav.test.tsx
Comment (firestack) [reply assets/tests/components/nav/topNav.test.tsx]: np! just wondering, was the removal of `expect(logoutLink).toBeVisible()` intentional?

Comment (lemald) [reply assets/tests/components/nav/topNav.test.tsx]: Oops, it was not! 0c9f91be

File: assets/tests/components/nav/topNav.test.tsx
Comment (firestack) [inline assets/tests/components/nav/topNav.test.tsx:75]:
```suggestion
    test("brings up an element with 'logged in as' text and a logout link when clicked", async () => {
```

File: assets/src/components/nav/topNav.tsx
Comment (joshlarson) [inline assets/src/components/nav/topNav.tsx:55]: Non-blocking Nit: 👇 

```suggestion
                    show
```

Since the popover's visibility is already governed by `showUserPopover`, I don't think we also need to have the menu itself be governed that way.

And this makes it easier to extract the `Dropdown.Menu` part to a separate component if we wanted to have it live in Storybook.

File: assets/tests/components/nav/topNav.test.tsx
Comment (joshlarson) [inline assets/tests/components/nav/topNav.test.tsx:90]: **praise**: Appreciate replacing `result` with `screen` here! I think I goofed that when I added these tests, and this is helpful to give me a better sense of when `screen` is usable (I'm now thinking more like... always 🤯)

File: assets/src/components/nav/topNav.tsx
Comment (firestack) [inline assets/src/components/nav/topNav.tsx:68]: question/suggestion: what is this `<span/>` doing? Could this be replaced with `icon-link`?
```suggestion
                      className="w-auto icon-link"
                    >
                      <BsIcon.BoxArrowRight className="me-2" /> Log out
```

File: assets/src/components/nav/topNav.tsx
Comment (lemald) [reply assets/src/components/nav/topNav.tsx:55]: Good catch, thanks!

**Review by firestack — COMMENTED** [review]
I still need an answer for https://github.com/mbta/skate/pull/2395/files#r1489930450 and I'm waiting on a reply for https://github.com/mbta/skate/pull/2395/files#r1489988986 but this looks good otherwise!

File: assets/src/components/nav/topNav.tsx
Comment (lemald) [reply assets/src/components/nav/topNav.tsx:68]: Yes, good call. I'm not going to commit this suggestion directly because it turns out I had to tweak some other styling and utility classes as well to make everything work out (though those changes ended up fixing another little annoying style detail I was trying to debug).

Comment (lemald) [reply assets/src/components/nav/topNav.tsx]: 1dc9141b

File: assets/css/nav/_top_nav.scss
Comment (lemald) [reply assets/css/nav/_top_nav.scss:67]: > I don't think it's worth guessing what we'll do in the future though, it's probably better to stick to defaults and wait for the reboot/typo/colors work to be done.

Yeah unless design specifically requests it I think this is the safer option for now.

File: assets/src/components/nav/topNav.tsx
Comment (firestack) [inline assets/src/components/nav/topNav.tsx:64]: (it seems like this is still in flux, I'm just noting this so it doesn't get lost)
nit(non-blocking): I'm not seeing `w-100` do anything when I remove it? 

side-note: I learned recently that `width: auto` is the default behaviour of `width` (which `auto` essentially turns into `100%`), and often `width: 100%` is redundant, so I'm wondering if that applies here.

Comment (lemald) [reply assets/src/components/nav/topNav.tsx:64]: Ah, I think that and the `w-auto` are vestigial then, I'll remove them.

File: assets/css/nav/_top_nav.scss
Comment (firestack) [reply assets/css/nav/_top_nav.scss:67]: But I still need the resetting of borders to `0` on `hr`, regardless of the color. could that be moved into the global namespace, or have `c-nav-menu` added to the list of allowed parents on this selector?

I could otherwise open a PR to add this for just the `c-nav-menu` but then we need to remove the `hr` styles in two places once we have the reboot.

Comment (lemald) [reply assets/css/nav/_top_nav.scss:67]: Yes, sorry, I'm working on some other stuff and haven't gotten to addressing that part but it's on my mind.

Comment (firestack) [reply assets/css/nav/_top_nav.scss:67]: Okay, thanks!!

File: assets/src/components/nav/topNav.tsx
Comment (lemald) [reply assets/src/components/nav/topNav.tsx]: 9ddc5815

**Comment by joshlarson** [comment]
I'm holding of on approving because of @firestack's out-standing comments, but once those are addressed and she approves, then I'm also good to go.

File: assets/css/nav/_top_nav.scss
Comment (lemald) [reply assets/css/nav/_top_nav.scss:67]: As for the border reset stuff... I'm kind of struggling to identify anywhere that would be good to put it without potentially impacting other `<hr/>` elements we already have, though I'm open to suggestions.

Comment (firestack) [reply assets/css/nav/_top_nav.scss:67]: What if we created a `bootstrap_polyfills` or something and then added this rule?

```
.c-top-nav__popover, .c-nav-menu {
  hr {
    border-left: 0;
    border-right: 0;
    border-bottom: 0;
  }
}
```

Comment (lemald) [reply assets/css/nav/_top_nav.scss:67]: Oh, I like that, and it makes it very clear what it's there for so that we know to remove it as we add more Bootstrap stuff! How about I add it as `css/bootstrap/_polyfills.scss` to work it into our existing folder structure?

Comment (lemald) [reply assets/css/nav/_top_nav.scss]: Well I went ahead and did it, let me know what you think: 6a20a8bd

Comment (firestack) [reply assets/css/nav/_top_nav.scss]: Love it!

**Review by firestack — APPROVED** [review]

**Pull request merged**
